tofhir-server/src/main/scala/io/tofhir/server/service/schema/SchemaFolderRepository.scala

```diff
@@ -354,5 +334,92 @@ class SchemaFolderRepository(schemaRepositoryFolderPath: String, projectFolderRe
     }
     baseFhirConfig
   }
+
+  /**
+   * @param projectId Project containing the schema definition
+   * @param id        Identifier of the schema definition
+   * @return          Structure definition of the schema
+   */
+  override def getSchemaAsStructureDefinition(projectId: String, id: String): Future[Option[Resource]] = {
+    getSchema(projectId, id).map {
+      case Some(schemaStructureDefinition) =>
+        Some(SchemaUtil.convertToStructureDefinitionResource(schemaStructureDefinition))
+      case None =>
+        None
+    }
+  }
+
+  /**
+   * Save a schema by using directly its structure definition resource
+   * @param projectId
+   * @param structureDefinitionResource Structure definition resource of the schema
+   *  @return
+   */
+  override def saveSchemaByStructureDefinition(projectId: String, structureDefinitionResource: Resource): Future[SchemaDefinition] = {
+    // Validate the resource
+    try {
+      fhirConfigurator.validateGivenInfrastructureResources(baseFhirConfig, api.FHIR_FOUNDATION_RESOURCES.FHIR_STRUCTURE_DEFINITION, Seq(structureDefinitionResource))
+    } catch {
+      case e: Exception =>
+        throw BadRequest("Schema resource is not valid.", s"Schema resource cannot be validated.", Some(e))
+    }
+
+    // Create structureDefinition from the resource
+    val structureDefinition: ProfileRestrictions = fhirFoundationResourceParser.parseStructureDefinition(structureDefinitionResource, includeElementMetadata = true)
+    // Generate an Id if id is missing
+    val schemaId = structureDefinition.id.getOrElse(UUID.randomUUID().toString)
+
+    checkIfSchemaIsUnique(projectId, schemaId, structureDefinition.url)
+
+    // To use convertToSchemaDefinition, profileRestrictions sequence must include the structure definition. Add it before conversion
+    baseFhirConfig.profileRestrictions += structureDefinition.url -> structureDefinition
+    val schemaDefinition = convertToSchemaDefinition(structureDefinition, simpleStructureDefinitionService)
+    // Remove structure definition from the cache and add it after file writing is done to ensure files and cache are the same
+    baseFhirConfig.profileRestrictions -= structureDefinition.url
+
+    // Write to the repository as a new file and update caches
+    writeSchemaAndUpdateCaches(projectId, structureDefinitionResource, schemaDefinition)
+  }
+
+  /**
+   * Checks if ID of the schema is unique in the project and url is unique in the program
+   * @param projectId
```

Comment:
Complete the JavaDoc for parameters' definition and describe the possible exceptions.